File: py/solns/naryTreePreorderTraversal/naryTreePreorderTraversal.py
class Node:
    def __init__(self,val=None,children=None):
        self.val = val
        self.children = children if children is not None else []

class Solution:
    @staticmethod
    def naive(root):
        res = []
        def preOrder(tree):
            if tree==None: return 
            res.append(tree.val)
            for child in tree.children:
                preOrder(child)
        preOrder(root)
        return res

File: cpp/example/src/NaryTreePreorderTraversal/naryTreePreorderTraversal.cpp
#include "naryTreePreorderTraversal.h"

void naryTreePreorderTraversal::recursive(Node* tree){
    if (tree==nullptr) return;
    this->res.push_back(tree->val);
    for (auto child: tree->children)
        recursive(child);
}

vector<int> naryTreePreorderTraversal::naive(Node* root){
    recursive(root);
    return this->res;
}
